File: src/SWEA1228/Solution.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.LinkedList;

// 암호문 - 1
public class Solution {
  public static void main(String[] args) throws IOException {
    final int TESTCASE_COUNT = 10;

    BufferedReader bReader = new BufferedReader(new InputStreamReader(System.in));
    BufferedWriter bWriter = new BufferedWriter(new OutputStreamWriter(System.out));
    
    for (int T = 1; T <= TESTCASE_COUNT; T++) {
      StringBuilder sBuilder = new StringBuilder();

      bReader.readLine(); // 원본 암호문의 길이 
      
      String[] input = bReader.readLine().split(" "); // 원본 암호문

      LinkedList<String> password = new LinkedList<>();
      for (String each : input) {
        password.add(each);
      }

      int commandCount = Integer.parseInt(bReader.readLine()); // 명령어의 개수

      String[] commands = bReader.readLine().split(" "); // 명령어
      
      int count = 0;
      int index = 0;
      while (count < commandCount) {
        index++; // I 알파벳 부분 넘기기
        int insertIndex = Integer.parseInt(commands[index++]);
        int insertCount = Integer.parseInt(commands[index++]);

        LinkedList<String> temp = new LinkedList<>();
        for (int i = 0; i < insertCount; i++) {
          if (insertIndex + i >= 10) {
            break;
          }
          temp.add(commands[index + i]);
        }
        if (insertIndex < 10) {
          password.addAll(insertIndex, temp);
        }
        
        count++;
        index += insertCount;
      }

      sBuilder.append(String.format("#%d ", T));
      for (int i = 0; i < 10; i++) {
        sBuilder.append(String.format("%s ", password.get(i)));
      }
      sBuilder.append("\n");
      bWriter.write(sBuilder.toString());
    }

    bWriter.flush();

    bReader.close();
    bWriter.close();
  }
}

File: src/SWEA1228/Solution.js
/**
 * @param {number[]} password
 * @param {number[]} commands
 * @return {number[]}
 */
const Solution = (password, commands) => {
  commands.forEach(each => {
    const [idx, numbers] = each;

    password.splice(idx, 0, ...numbers);
  });

  return password.slice(0, 10);
};

const password = [449047, 855428, 425117, 532416, 358612, 929816, 313459, 311433, 472478, 589139, 568205];

const commands = [
  [1, [400905, 139831, 966064, 336948, 119288]],
  [8, [436704, 702451, 762737, 557561, 810021, 771706]],
  [3, [389953, 706628, 552108, 238749, 661021, 498160, 493414, 377808]],
  [13, [237017, 301569, 243869, 252994]],
  [3, [408347, 618608, 822798, 370982]],
  [8, [424216, 356268]],
  [4, [512816, 992679, 693002, 835918, 768525, 949227, 628969, 521945, 839380, 479976]]
];

console.log(Solution(password, commands));